frontend/lib/resolution/Resolver.cpp

```diff
@@ -2990,6 +2990,15 @@ void Resolver::exit(const Call* call) {
 
     // handle type inference for variables split-inited by 'out' formals
     adjustTypesForOutFormals(ci, actualAsts, c.mostSpecific());
+  } else {
+    // For practical development/debugging, set the ResolvedExpression for
+    // this call in case resolution proceeds to the point where some other
+    // part of resolution assumes there is an entry for this call.
+    //
+    // Otherwise we would run into out_of_range exceptions when accessing
+    // the resolution result.
```

Comment:
I am surprised we have C++ exceptions at all (and would have expected assertion failures in that case). Nonetheless this change is OK. I could imagine it adds to the ResolutionResultByPostorderId storage size, but it seems likely that most Calls are resolve anyway, at least in normal operation.